Patcher/Program.cs

```diff
@@ -167,7 +186,8 @@ internal static void Main(string[] args)
 
             if (string.IsNullOrWhiteSpace(os))
             {
-                Console.WriteLine($"No OS was specified - please specify a valid operating system when running the patcher. See patcher -h for available options.");
+                Console.WriteLine(
```

Comment:
I found out that you can detect the OS in Dotnet Core using: `RuntimeInformation.IsOSPlatform`. It might make sense to use the OS which is running the Patcher as a default, but let people overwrite it using the respective flags. If you want I can create a separate PR for that

Reply:
I was thinking something along those lines earlier as well. If we can detect the OS the user is running, we can also guess where their Unity installs are located as well. If we simply print those folders, we can automatically detect and give the user the choice of which version to patch, or let them know that the patcher doesn't support their version of Unity.

All the while, of course, keeping all the manual switches and overrides in tact. If you want to have a hack at it, be my guest. I think it's a good idea to simplify the patcher that way.

Reply:
Before you start working on it, could you open an issue outlining the design and the changes you'd make? Gives us a place to discuss the changes a bit more before submitting a PR. :)

Reply:
I described what I had in mind in #28 feel free to tear it apart :)